src/neo/Consensus/ConsensusContext.cs

```diff
@@ -137,6 +158,20 @@ public Block EnsureHeader()
             return Block;
         }
 
+        public StateRoot EnsureStateRoot()
+        {
+            if (PreviousBlockStateRoot is null)
+            {
+                PreviousBlockStateRoot = new StateRoot
+                {
+                    Version = StateRoot.CurrentVersion,
+                    Index = Block.Index - 1,
+                    RootHash = Blockchain.Singleton.GetLocalStateRoot(Block.Index - 1)
```

Comment:
If we don't have the state synchronized , what's happend? we should not be able to be a CN, isn't it?

Reply:
I think it doesn't matter.
Because all consensus nodes vote for their calculated StateRoot. As long as the node block synchronization is completed, it has the ability to verify StateRoot, regardless of whether the State Root synchronization is completed. Even if the StateRoot is not synchronized, its StateRoot may be correct. If the StateRoot is incorrect, it will be rejected by other nodes.